### DISCUSSION TITLE
Pornhub и Adguard после обновления nfqws

### DISCUSSION THREAD

**Discussion by shizeggo1** [Anonym-tsk/nfqws-keenetic#106] [Q&A] [answered]
Здравствуйте. В общем, проблема такая: после недавнего обновления через

opkg update
opkg upgrade nfqws-keenetic
opkg upgrade nfqws-keenetic-web

сайт Порнхаба медленно открывается со включенным Адгуард (без Адгуард быстро). С прошлой версией всё было ок (в user.list прописаны нужные домены), но стал медленно работать Ютуб, потому и решил обновить nfqws.
Помогите разобраться, прикладываю старый конфиг (порнхаб+агдуард работает, ютуб тормозит):

```
# Provider network interface, e.g. eth3
# You can specify multiple interfaces separated by space, e.g. ISP_INTERFACE="eth2.2"
ISP_INTERFACE="eth2.2"

# All arguments here: https://github.com/bol-van/zapret (search for `nfqws` on the page)
# HTTP(S) strategy
NFQWS_ARGS="--filter-udp=50000-65535 --dpi-desync=fake --dpi-desync-any-protocol --dpi-desync-repeats=6 --new --dpi-desync=fake,split2 --dpi-desync-ttl=0 --dpi-desync-repeats=6 --dpi-desync-split-pos=1 --dpi-desync-fooling=md5sig,badseq --dpi-desync-fake-tls=/opt/etc/nfqws/tls_clienthello.bin"

# QUIC strategy
NFQWS_ARGS_QUIC="--dpi-desync=fake --dpi-desync-repeats=6 --dpi-desync-fooling=badsum --dpi-desync-ttl=0 --dpi-desync-fake-quic=/opt/etc/nfqws/quic_initial.bin"

# auto - automatically detects blocked resources and adds them to the auto.list
NFQWS_EXTRA_ARGS="--hostlist=/opt/etc/nfqws/user.list --hostlist-auto=/opt/etc/nfqws/auto.list --hostlist-auto-debug=/opt/var/log/nfqws.log --hostlist-exclude=/opt/etc/nfqws/exclude.list"

# list - applies rules only to domains in the user.list
#NFQWS_EXTRA_ARGS="--hostlist=/opt/etc/nfqws/user.list"

# all  - applies rules to all traffic except domains from exclude.list
#NFQWS_EXTRA_ARGS="--hostlist-exclude=/opt/etc/nfqws/exclude.list"

# IPv6 support
IPV6_ENABLED=0

# HTTP support
HTTP_ENABLED=0

# QUIC support
QUIC_ENABLED=1
UDP_PORTS=443,50000:65535

# Syslog logging level (0 - silent, 1 - debug)
LOG_LEVEL=0

NFQUEUE_NUM=200
USER=nobody
CONFIG_VERSION=3
```

Новый конфиг последней версии [v2.5.9] (адгуард+порнхаб тормозит, ютуб работает):

```
# Provider network interface, e.g. eth3
# You can specify multiple interfaces separated by space, e.g. ISP_INTERFACE="eth3 nwg1"
ISP_INTERFACE="eth2.2"

# All arguments here: https://github.com/bol-van/zapret (search for `nfqws` on the page)
# HTTP(S) strategy
NFQWS_ARGS="--dpi-desync=fake,split2 --dpi-desync-ttl=0 --dpi-desync-repeats=16 --dpi-desync-split-pos=1 --dpi-desync-fooling=md5sig,badseq --dpi-desync-fake-tls=/opt/etc/nfqws/tls_clienthello.bin"

# QUIC strategy
NFQWS_ARGS_QUIC="--filter-udp=443 --dpi-desync=fake --dpi-desync-repeats=11 --dpi-desync-fake-quic=/opt/etc/nfqws/quic_initial.bin"

# UDP strategy (doesn't use lists from NFQWS_EXTRA_ARGS)
NFQWS_ARGS_UDP="--filter-udp=50000-50099 --dpi-desync=fake --dpi-desync-any-protocol --dpi-desync-repeats=6 --dpi-desync-cutoff=n2"

# auto - automatically detects blocked resources and adds them to the auto.list
NFQWS_EXTRA_ARGS="--hostlist=/opt/etc/nfqws/user.list --hostlist-auto=/opt/etc/nfqws/auto.list --hostlist-auto-debug=/opt/var/log/nfqws.log --hostlist-exclude=/opt/etc/nfqws/exclude.list"

# list - applies rules only to domains in the user.list
#NFQWS_EXTRA_ARGS="--hostlist=/opt/etc/nfqws/user.list"

# all  - applies rules to all traffic except domains from exclude.list
#NFQWS_EXTRA_ARGS="--hostlist-exclude=/opt/etc/nfqws/exclude.list"

# IPv6 support
IPV6_ENABLED=0

# TCP ports for iptables rules
TCP_PORTS=443

# UDP ports for iptables rules
UDP_PORTS=443,50000:50099

# Syslog logging level (0 - silent, 1 - debug)
LOG_LEVEL=0

NFQUEUE_NUM=200
USER=nobody
CONFIG_VERSION=4

```

**Comment by pumi** [comment]
Добавьте в auto.list или в user.list домен (рестарт потом лучше сделать скрипта):
phncdn.com
pornhub.com

я так понял phncdn.com это видео pornhub. Должно заработать. Способ гуглится, оказывается с августа замедляют как ютуб =) вот например https://ntc.party/t/%D0%BE%D0%B1%D1%85%D0%BE%D0%B4-%D0%B7%D0%B0%D0%BC%D0%B5%D0%B4%D0%BB%D0%B5%D0%BD%D0%B8%D1%8F-pornhub/8293/2 
п.с. там даже советуют еще побольше добавить доменов
pornhub.com
rt.pornhub.com
ei.phncdn.com
phncdn.com
ew.phncdn.com
ss.phncdn.com
cdn1d-static-shared.phncdn.com
 Хотя моё мнение это лишнее поддомены добавлять, если сделать только pornhub.com и phncdn.com то будут и поддомены работать.

**Reply by shizeggo1** [reply]
Домены были добавлены, проблема остается. Что любопытно, с выключенным Адгардом сайт грузится мгновенно, то есть сам сайт замедляет работу с ним при включенном адблоке. Пробовал разные сценарии, так и не разобрался толком.

**Reply by pumi** [reply]
Adguard это плагин для браузера, программа в винде или в роутере установлен?

**Reply by shizeggo1** [reply]
Программа в винде.

**Reply by pumi** [reply]
Там скорее всего как-то прописывается DNS-сервер адгвард и мешает нормальной работе? На твиттер, фэйсбук, rutracker заходит при включенном?

**Reply by shizeggo1** [reply]
DNS стоит "использовать вашего интернет-провайдера". Со старым конфигом не заходит на рутрекер, с новым - заходит.

**Reply by pumi** [reply]
А в инструкции перед установкой (на роутере) вот так:
Рекомендуется игнорировать предложенные провайдером адреса DNS-серверов. Для этого в интерфейсе роутера отметьте пункты ["игнорировать DNS от провайдера"](https://help.keenetic.com/hc/ru/articles/360008609399) в настройках IPv4 и IPv6.

Вместе с этим рекомендуется [настроить использование DoT/DoH](https://help.keenetic.com/hc/ru/articles/360007687159).

Ну и вообще надо разобраться что там адгвард с траффиком делает, куда его "гоняет" для "блокировки рекламы" =)

**Reply by shizeggo1** [reply]
Ну, всё, в чем я уверен - со старым конфигом Адгуард+порнхаб и половина Ютуба работает, с новыми - Ютуб работает, а именно сайту Порнхаб не нравится наличие адблока, и сайт работает медленно. То есть раньше (со старым конфигом) получалось дурить Порнхаб, а с новым конфигом - нет. Нужно что-то куда-то прописать в конфиге, а что - так и не разобрался. Пробовал в разные "стратегии" через --new и custom.list, как в мануале написано - всё перепробовал, либо ютуб отваливается, либо порнхаб, либо вообще всё. Вряд ли с DNS тут проблема.

**Reply by pumi** [reply]
Ну может быть и что-то изменилось в конфиге и надо добавить параметры. Попробуйте взять параметры из старого конфига где всё работало и как-то доработать, чтобы и ютуб и пхаб работал. 
Ещё может быть вариант, что провайдер что-то изменил у себя тоже и стало вот так.

**Comment by shizeggo1** [comment, marked as answer]
Похоже, нашел решение: добавил --dpi-desync-fooling=badsum в NFQWS_ARGS и NFQWS_ARGS_QUIC
